frontend/script.js

```diff
@@ -0,0 +1,57 @@
+function showSideBar() {
+    const sidebar = document.querySelector('.hamburger');
+    sidebar.style.display = 'flex';
+}
+
+function hideSideBar() {
+    const sidebar = document.querySelector('.hamburger');
+    sidebar.style.display = 'none';
+}
+
+// Specify the API endpoint for user data
+const apiUrl = 'http://localhost:8080/products/';
+// Make a GET request using the Fetch API
+fetch(apiUrl)
+    .then((response) => {
+        if (!response.ok) {
+            throw new Error('Network response was not ok');
+        }
+        return response.json();
+    })
+    .then((data) => {
+        data.data.forEach((element) => {
+            if ('content' in document.createElement('template')) {
+                const trendingBody = document.querySelector('#trending');
+                const template = document.querySelector('#trending-product');
+                const clone = template.content.cloneNode(true);
+                let tp = clone.querySelectorAll('div');
+                const img = tp[0].querySelector('img');
+                img.addEventListener('error', function (event) {
+                    event.target.src = './assets/images/trending-product.webp';
+                    event.onerror = null;
+                });
+                tp[1].textContent = element.name;
+                tp[2].textContent = '£' + element.unitPrice;
+                trendingBody.appendChild(clone);
+            } else {
+                tp[0].textContent = 'Placeholder';
+                tp[1].textContent = '£XX.XX';
+            }
+        });
+    })
+    .catch((error) => {
+        console.error('Error:', error);
+
+        for (let i = 0; i < 4; i++) {
+            if ('content' in document.createElement('template')) {
```

Comment:
Can this whole expression be separated to a variable or function, explaining what it does?

Reply:
Code has been moved in to a function called handleError which sits within the Product object